src/GeneratedInjectorDelegator.php

```diff
@@ -0,0 +1,38 @@
+<?php
+/**
+ * @see       https://github.com/zendframework/zend-di for the canonical source repository
+ * @copyright Copyright (c) 2019 Zend Technologies USA Inc. (https://www.zend.com)
+ * @license   https://github.com/zendframework/zend-di/blob/master/LICENSE.md New BSD License
+ */
+
+namespace Zend\Di;
+
+use Psr\Container\ContainerInterface;
+use Zend\Di\Exception\InvalidServiceConfigException;
+use function class_exists;
+
+class GeneratedInjectorDelegator
+{
+    /**
+     * @param string $name
+     */
+    public function __invoke(ContainerInterface $container, $name, callable $callback) : InjectorInterface
+    {
+        $config = $container->has('config') ? $container->get('config') : [];
+        $aotConfig = $config['dependencies']['auto']['aot'] ?? [];
+        $namespace = empty($aotConfig['namespace']) ? 'Zend\Di\Generated' : $aotConfig['namespace'];
```

Comment:
Maybe better "empty or non-string" here?

Reply:
```suggestion
        $namespace = empty($aotConfig['namespace']) || !is_string($aotConfig['namespace']) 
            ? 'Zend\Di\Generated' 
            : $aotConfig['namespace'];
```

Like this?

Reply:
yes, please

test would be with an array, for example, as it can't be casted to string

Reply:
I was just going to make this change and saw it would render the next 3 lines useless:

```php
        $config = $container->has('config') ? $container->get('config') : [];
        $aotConfig = $config['dependencies']['auto']['aot'] ?? [];
        $namespace = empty($aotConfig['namespace']) ? 'Zend\Di\Generated' : $aotConfig['namespace'];

        if (! is_string($namespace)) {
            throw new InvalidServiceConfigException('Provided namespace is not a string.');
        }
```

I'd argue to leave it as is. This will be explicit to the user that there is an unusable misconfiguration instead of hiding it behind a default value that may cause failures which may become hard to debug. At least it's not obvious what's the issue that cause incorrect behavior.

I'll change the test from an int fixture to an array though.